Library/Homebrew/sbom.rb

```diff
@@ -0,0 +1,354 @@
+          "name"             => dependency["name"],
+          "versionInfo"      => dependency["pkg_version"],
+          "filesAnalyzed"    => false,
+          "licenseDeclared"  => "NOASSERTION",
+          "licenseConcluded" => dependency["license"] || "NOASSERTION",
+          "downloadLocation" => bottle_info.present? ? bottle_info["url"] : "NOASSERTION",
+          "copyrightText"    => "NOASSERTION",
+          "checksums"        => [
+            {
+              "algorithm"     => "SHA256",
+              "checksumValue" => bottle_info.present? ? bottle_info["sha256"] : "NOASSERTION",
+            },
+          ],
+          "externalRefs"     => [
+            {
+              "referenceCategory" => "PACKAGE-MANAGER",
+              "referenceLocator"  => "pkg:brew/#{dependency["full_name"]}@#{dependency["version"]}",
+              "referenceType"     => "purl",
+            },
+          ],
+        }
+      end
+    end
+
+    compiler_info = {
+      "SPDXRef-Compiler" => {
+        "SPDXID"           => "SPDXRef-Compiler",
+        "name"             => compiler.to_s,
+        "versionInfo"      => T.unsafe(built_on["xcode"]) || "NOASSERTION",
+        "filesAnalyzed"    => false,
+        "licenseDeclared"  => "NOASSERTION",
+        "licenseConcluded" => "NOASSERTION",
+        "copyrightText"    => "NOASSERTION",
+        "downloadLocation" => "NOASSERTION",
+        "checksums"        => [],
+        "externalRefs"     => [],
+      },
+    }
+
+    if stdlib.present?
+      compiler_info["SPDXRef-Stdlib"] = {
+        "SPDXID"           => "SPDXRef-Stdlib",
+        "name"             => stdlib,
+        "versionInfo"      => stdlib,
+        "filesAnalyzed"    => false,
+        "licenseDeclared"  => "NOASSERTION",
+        "licenseConcluded" => "NOASSERTION",
+        "copyrightText"    => "NOASSERTION",
+        "downloadLocation" => "NOASSERTION",
+        "checksums"        => [],
+        "externalRefs"     => [],
+      }
+    end
+
+    packages = generate_packages_json(runtime_full, compiler_info)
+    {
+      "SPDXID"            => "SPDXRef-DOCUMENT",
+      "spdxVersion"       => "SPDX-2.3",
+      "name"              => "SBOM-SPDX-#{name}-#{stable_version}",
+      "creationInfo"      => {
+        "created"  => DateTime.now.to_s,
+        "creators" => ["Tool: https://github.com/homebrew/brew@#{homebrew_version}"],
+      },
+      "dataLicense"       => "CC0-1.0",
+      "documentNamespace" => "https://formulae.brew.sh/spdx/#{name}-#{stable_version}.json",
+      "documentDescribes" => packages.map { |dependency| dependency["SPDXID"] },
+      "files"             => [],
+      "packages"          => packages,
+      "relationships"     => generate_relations_json(runtime_full, compiler_info),
+    }
+  end
+
+  sig { params(formula: Formula, deps: T::Array[Dependency]).returns(T::Array[Hash]) }
+  def self.runtime_deps_hash(formula, deps)
+    deps.map do |dep|
+      f = dep.to_formula
```

Comment:
Would be good to have test coverage to cover this

Reply:
This is showing up as not covered again. Unsure if Codecov is lying, comment and resolve if so (reproduced with `brew tests --coverage`).

Reply:
I'm not sure either. I do have a test with runtime dependencies, so I would expect this to be covered. Not sure how to check this exactly

Reply:
@SMillerDev can always confirm by putting a `raise` in here and then running `brew tests`. If it doesn't go 💥: it wasn't covered.

Reply:
That is a remarkably easy way to detect that 😅